gguf-py/gguf/constants.py

```diff
@@ -241,59 +274,88 @@ class MODEL_TENSOR(IntEnum):
+    MODEL_TENSOR.FFN_NORM:             "blk.{bid}.ffn_norm",
+    MODEL_TENSOR.FFN_GATE:             "blk.{bid}.ffn_gate",
+    MODEL_TENSOR.FFN_DOWN:             "blk.{bid}.ffn_down",
+    MODEL_TENSOR.FFN_UP:               "blk.{bid}.ffn_up",
+    MODEL_TENSOR.FFN_GATE_SHEXP:       "blk.{bid}.ffn_gate_shexp",
+    MODEL_TENSOR.FFN_DOWN_SHEXP:       "blk.{bid}.ffn_down_shexp",
+    MODEL_TENSOR.FFN_UP_SHEXP:         "blk.{bid}.ffn_up_shexp",
+    MODEL_TENSOR.FFN_ACT:              "blk.{bid}.ffn",
+    MODEL_TENSOR.FFN_NORM_EXP:         "blk.{bid}.ffn_norm_exps",
+    MODEL_TENSOR.FFN_GATE_EXP:         "blk.{bid}.ffn_gate_exps",
+    MODEL_TENSOR.FFN_DOWN_EXP:         "blk.{bid}.ffn_down_exps",
+    MODEL_TENSOR.FFN_UP_EXP:           "blk.{bid}.ffn_up_exps",
+    MODEL_TENSOR.LAYER_OUT_NORM:       "blk.{bid}.layer_output_norm",
+    MODEL_TENSOR.SSM_IN:               "blk.{bid}.ssm_in",
+    MODEL_TENSOR.SSM_CONV1D:           "blk.{bid}.ssm_conv1d",
+    MODEL_TENSOR.SSM_X:                "blk.{bid}.ssm_x",
+    MODEL_TENSOR.SSM_DT:               "blk.{bid}.ssm_dt",
+    MODEL_TENSOR.SSM_A:                "blk.{bid}.ssm_a",
+    MODEL_TENSOR.SSM_D:                "blk.{bid}.ssm_d",
+    MODEL_TENSOR.SSM_OUT:              "blk.{bid}.ssm_out",
+    MODEL_TENSOR.ATTN_Q_A:             "blk.{bid}.attn_q_a",
+    MODEL_TENSOR.ATTN_Q_B:             "blk.{bid}.attn_q_b",
+    MODEL_TENSOR.ATTN_KV_A_MQA:        "blk.{bid}.attn_kv_a_mqa",
+    MODEL_TENSOR.ATTN_KV_B:            "blk.{bid}.attn_kv_b",
+    MODEL_TENSOR.ATTN_Q_A_NORM:        "blk.{bid}.attn_q_a_norm",
+    MODEL_TENSOR.ATTN_KV_A_NORM:       "blk.{bid}.attn_kv_a_norm",
+    MODEL_TENSOR.ATTN_SUB_NORM:        "blk.{bid}.attn_sub_norm",
+    MODEL_TENSOR.FFN_SUB_NORM:         "blk.{bid}.ffn_sub_norm",
+    MODEL_TENSOR.DEC_ATTN_NORM:        "dec.blk.{bid}.attn_norm",
+    MODEL_TENSOR.DEC_ATTN_Q:           "dec.blk.{bid}.attn_q",
+    MODEL_TENSOR.DEC_ATTN_K:           "dec.blk.{bid}.attn_k",
+    MODEL_TENSOR.DEC_ATTN_V:           "dec.blk.{bid}.attn_v",
+    MODEL_TENSOR.DEC_ATTN_OUT:         "dec.blk.{bid}.attn_o",
+    MODEL_TENSOR.DEC_ATTN_REL_B:       "dec.blk.{bid}.attn_rel_b",
+    MODEL_TENSOR.DEC_CROSS_ATTN_NORM:  "dec.blk.{bid}.cross_attn_norm",
+    MODEL_TENSOR.DEC_CROSS_ATTN_Q:     "dec.blk.{bid}.cross_attn_q",
+    MODEL_TENSOR.DEC_CROSS_ATTN_K:     "dec.blk.{bid}.cross_attn_k",
+    MODEL_TENSOR.DEC_CROSS_ATTN_V:     "dec.blk.{bid}.cross_attn_v",
+    MODEL_TENSOR.DEC_CROSS_ATTN_OUT:   "dec.blk.{bid}.cross_attn_o",
+    MODEL_TENSOR.DEC_CROSS_ATTN_REL_B: "dec.blk.{bid}.cross_attn_rel_b",
+    MODEL_TENSOR.DEC_FFN_NORM:         "dec.blk.{bid}.ffn_norm",
+    MODEL_TENSOR.DEC_FFN_GATE:         "dec.blk.{bid}.ffn_gate",
+    MODEL_TENSOR.DEC_FFN_DOWN:         "dec.blk.{bid}.ffn_down",
+    MODEL_TENSOR.DEC_FFN_UP:           "dec.blk.{bid}.ffn_up",
+    MODEL_TENSOR.DEC_OUTPUT_NORM:      "dec.output_norm",
+    MODEL_TENSOR.ENC_ATTN_NORM:        "enc.blk.{bid}.attn_norm",
```

Comment:
The `enc` and `dec` prefixes will (eventually) need to be also handled by the new markdown output mode of `gguf-dump.py` (#7853).

Can be fixed in a separate PR, I'm mentioning this for future reference.

(@mofosyne, you should be aware of this)

Reply:
@compilade I tried it on one example model `python3 gguf-py/scripts/gguf-dump.py --markdown /mnt/md0/models/t5-small.gguf` and I'm not sure what could be fixed, can you be more specific?

| T_ID | Tensor Layer Name                 | Human Friendly Tensor Layer Name                   | Elements         | Shape                | Type |
|-----:|:----------------------------------|:---------------------------------------------------|:-----------------|:---------------------|:-----|
|    0 | dec.blk.0.attn_k.weight           | Dec Block 0 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    1 | dec.blk.0.attn_o.weight           | Dec Block 0 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    2 | dec.blk.0.attn_q.weight           | Dec Block 0 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    3 | dec.blk.0.attn_rel_b.weight       | Dec Block 0 Attn_Rel_B (W)                         | (  256)      256 |    8 x    32 x 1 x 1 | F16  |
|    4 | dec.blk.0.attn_v.weight           | Dec Block 0 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    5 | dec.blk.0.attn_norm.weight        | Dec Block 0 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|    6 | dec.blk.0.cross_attn_k.weight     | Dec Block 0 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    7 | dec.blk.0.cross_attn_o.weight     | Dec Block 0 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    8 | dec.blk.0.cross_attn_q.weight     | Dec Block 0 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|    9 | dec.blk.0.cross_attn_rel_b.weight | Dec Block 0 Cross_Attn_Rel_B (W)                   | (  256)      256 |    8 x    32 x 1 x 1 | F16  |
|   10 | dec.blk.0.cross_attn_v.weight     | Dec Block 0 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   11 | dec.blk.0.cross_attn_norm.weight  | Dec Block 0 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   12 | dec.blk.0.ffn_up.weight           | Dec Block 0 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   13 | dec.blk.0.ffn_down.weight         | Dec Block 0 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   14 | dec.blk.0.ffn_norm.weight         | Dec Block 0 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   15 | dec.blk.1.attn_k.weight           | Dec Block 1 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   16 | dec.blk.1.attn_o.weight           | Dec Block 1 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   17 | dec.blk.1.attn_q.weight           | Dec Block 1 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   18 | dec.blk.1.attn_v.weight           | Dec Block 1 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   19 | dec.blk.1.attn_norm.weight        | Dec Block 1 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   20 | dec.blk.1.cross_attn_k.weight     | Dec Block 1 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   21 | dec.blk.1.cross_attn_o.weight     | Dec Block 1 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   22 | dec.blk.1.cross_attn_q.weight     | Dec Block 1 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   23 | dec.blk.1.cross_attn_v.weight     | Dec Block 1 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   24 | dec.blk.1.cross_attn_norm.weight  | Dec Block 1 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   25 | dec.blk.1.ffn_up.weight           | Dec Block 1 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   26 | dec.blk.1.ffn_down.weight         | Dec Block 1 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   27 | dec.blk.1.ffn_norm.weight         | Dec Block 1 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   28 | dec.blk.2.attn_k.weight           | Dec Block 2 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   29 | dec.blk.2.attn_o.weight           | Dec Block 2 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   30 | dec.blk.2.attn_q.weight           | Dec Block 2 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   31 | dec.blk.2.attn_v.weight           | Dec Block 2 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   32 | dec.blk.2.attn_norm.weight        | Dec Block 2 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   33 | dec.blk.2.cross_attn_k.weight     | Dec Block 2 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   34 | dec.blk.2.cross_attn_o.weight     | Dec Block 2 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   35 | dec.blk.2.cross_attn_q.weight     | Dec Block 2 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   36 | dec.blk.2.cross_attn_v.weight     | Dec Block 2 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   37 | dec.blk.2.cross_attn_norm.weight  | Dec Block 2 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   38 | dec.blk.2.ffn_up.weight           | Dec Block 2 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   39 | dec.blk.2.ffn_down.weight         | Dec Block 2 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   40 | dec.blk.2.ffn_norm.weight         | Dec Block 2 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   41 | dec.blk.3.attn_k.weight           | Dec Block 3 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   42 | dec.blk.3.attn_o.weight           | Dec Block 3 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   43 | dec.blk.3.attn_q.weight           | Dec Block 3 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   44 | dec.blk.3.attn_v.weight           | Dec Block 3 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   45 | dec.blk.3.attn_norm.weight        | Dec Block 3 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   46 | dec.blk.3.cross_attn_k.weight     | Dec Block 3 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   47 | dec.blk.3.cross_attn_o.weight     | Dec Block 3 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   48 | dec.blk.3.cross_attn_q.weight     | Dec Block 3 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   49 | dec.blk.3.cross_attn_v.weight     | Dec Block 3 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   50 | dec.blk.3.cross_attn_norm.weight  | Dec Block 3 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   51 | dec.blk.3.ffn_up.weight           | Dec Block 3 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   52 | dec.blk.3.ffn_down.weight         | Dec Block 3 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   53 | dec.blk.3.ffn_norm.weight         | Dec Block 3 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   54 | dec.blk.4.attn_k.weight           | Dec Block 4 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   55 | dec.blk.4.attn_o.weight           | Dec Block 4 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   56 | dec.blk.4.attn_q.weight           | Dec Block 4 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   57 | dec.blk.4.attn_v.weight           | Dec Block 4 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   58 | dec.blk.4.attn_norm.weight        | Dec Block 4 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   59 | dec.blk.4.cross_attn_k.weight     | Dec Block 4 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   60 | dec.blk.4.cross_attn_o.weight     | Dec Block 4 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   61 | dec.blk.4.cross_attn_q.weight     | Dec Block 4 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   62 | dec.blk.4.cross_attn_v.weight     | Dec Block 4 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   63 | dec.blk.4.cross_attn_norm.weight  | Dec Block 4 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   64 | dec.blk.4.ffn_up.weight           | Dec Block 4 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   65 | dec.blk.4.ffn_down.weight         | Dec Block 4 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   66 | dec.blk.4.ffn_norm.weight         | Dec Block 4 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   67 | dec.blk.5.attn_k.weight           | Dec Block 5 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   68 | dec.blk.5.attn_o.weight           | Dec Block 5 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   69 | dec.blk.5.attn_q.weight           | Dec Block 5 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   70 | dec.blk.5.attn_v.weight           | Dec Block 5 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   71 | dec.blk.5.attn_norm.weight        | Dec Block 5 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   72 | dec.blk.5.cross_attn_k.weight     | Dec Block 5 Cross_Attn_K (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   73 | dec.blk.5.cross_attn_o.weight     | Dec Block 5 Cross_Attn_O (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   74 | dec.blk.5.cross_attn_q.weight     | Dec Block 5 Cross_Attn_Q (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   75 | dec.blk.5.cross_attn_v.weight     | Dec Block 5 Cross_Attn_V (W)                       | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   76 | dec.blk.5.cross_attn_norm.weight  | Dec Block 5 Cross_Attn_Norm (W)                    | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   77 | dec.blk.5.ffn_up.weight           | Dec Block 5 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   78 | dec.blk.5.ffn_down.weight         | Dec Block 5 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   79 | dec.blk.5.ffn_norm.weight         | Dec Block 5 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   80 | dec.output_norm.weight            | Dec Output Normalization (W)                       | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   81 | enc.blk.0.attn_k.weight           | Enc Block 0 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   82 | enc.blk.0.attn_o.weight           | Enc Block 0 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   83 | enc.blk.0.attn_q.weight           | Enc Block 0 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   84 | enc.blk.0.attn_rel_b.weight       | Enc Block 0 Attn_Rel_B (W)                         | (  256)      256 |    8 x    32 x 1 x 1 | F16  |
|   85 | enc.blk.0.attn_v.weight           | Enc Block 0 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   86 | enc.blk.0.attn_norm.weight        | Enc Block 0 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   87 | enc.blk.0.ffn_up.weight           | Enc Block 0 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   88 | enc.blk.0.ffn_down.weight         | Enc Block 0 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   89 | enc.blk.0.ffn_norm.weight         | Enc Block 0 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   90 | enc.blk.1.attn_k.weight           | Enc Block 1 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   91 | enc.blk.1.attn_o.weight           | Enc Block 1 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   92 | enc.blk.1.attn_q.weight           | Enc Block 1 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   93 | enc.blk.1.attn_v.weight           | Enc Block 1 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   94 | enc.blk.1.attn_norm.weight        | Enc Block 1 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   95 | enc.blk.1.ffn_up.weight           | Enc Block 1 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|   96 | enc.blk.1.ffn_down.weight         | Enc Block 1 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|   97 | enc.blk.1.ffn_norm.weight         | Enc Block 1 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|   98 | enc.blk.2.attn_k.weight           | Enc Block 2 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|   99 | enc.blk.2.attn_o.weight           | Enc Block 2 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  100 | enc.blk.2.attn_q.weight           | Enc Block 2 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  101 | enc.blk.2.attn_v.weight           | Enc Block 2 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  102 | enc.blk.2.attn_norm.weight        | Enc Block 2 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  103 | enc.blk.2.ffn_up.weight           | Enc Block 2 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|  104 | enc.blk.2.ffn_down.weight         | Enc Block 2 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|  105 | enc.blk.2.ffn_norm.weight         | Enc Block 2 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  106 | enc.blk.3.attn_k.weight           | Enc Block 3 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  107 | enc.blk.3.attn_o.weight           | Enc Block 3 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  108 | enc.blk.3.attn_q.weight           | Enc Block 3 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  109 | enc.blk.3.attn_v.weight           | Enc Block 3 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  110 | enc.blk.3.attn_norm.weight        | Enc Block 3 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  111 | enc.blk.3.ffn_up.weight           | Enc Block 3 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|  112 | enc.blk.3.ffn_down.weight         | Enc Block 3 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|  113 | enc.blk.3.ffn_norm.weight         | Enc Block 3 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  114 | enc.blk.4.attn_k.weight           | Enc Block 4 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  115 | enc.blk.4.attn_o.weight           | Enc Block 4 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  116 | enc.blk.4.attn_q.weight           | Enc Block 4 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  117 | enc.blk.4.attn_v.weight           | Enc Block 4 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  118 | enc.blk.4.attn_norm.weight        | Enc Block 4 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  119 | enc.blk.4.ffn_up.weight           | Enc Block 4 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|  120 | enc.blk.4.ffn_down.weight         | Enc Block 4 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|  121 | enc.blk.4.ffn_norm.weight         | Enc Block 4 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  122 | enc.blk.5.attn_k.weight           | Enc Block 5 Attention Key (W)                      | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  123 | enc.blk.5.attn_o.weight           | Enc Block 5 Attn_O (W)                             | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  124 | enc.blk.5.attn_q.weight           | Enc Block 5 Attention Query (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  125 | enc.blk.5.attn_v.weight           | Enc Block 5 Attention Value (W)                    | (~262K)   262144 |  512 x   512 x 1 x 1 | F16  |
|  126 | enc.blk.5.attn_norm.weight        | Enc Block 5 Attention Normalization (W)            | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  127 | enc.blk.5.ffn_up.weight           | Enc Block 5 Feed-Forward Network "Up" (W)          | (  ~1M)  1048576 |  512 x  2048 x 1 x 1 | F16  |
|  128 | enc.blk.5.ffn_down.weight         | Enc Block 5 Feed-Forward Network "Down" (W)        | (  ~1M)  1048576 | 2048 x   512 x 1 x 1 | F16  |
|  129 | enc.blk.5.ffn_norm.weight         | Enc Block 5 Feed-Forward Network Normalization (W) | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  130 | enc.output_norm.weight            | Enc Output Normalization (W)                       | (  512)      512 |  512 x     1 x 1 x 1 | F32  |
|  131 | token_embd.weight                 | Token Embedding (W)                                | ( ~16M) 16449536 |  512 x 32128 x 1 x 1 | F16  |

Reply:
There's currently a special case for tensor names which ***don't*** start with `blk`, and it puts them all in the same section (so that `token_embd.weight` is in the same section as `output.weight`). If you try it on a non-T5 model (e.g. tinyllama or something), you'll notice that there are sections for each layer number.

Reply:
Fixed in #8090